### PR TITLE
fix: allow ESM preset to be used in monorepo structure

### PR DESCRIPTION
**Pull request by lacolaco**
## Summary

Because `<rootDir>` is not workspace-root directory when it has a monorepo structure like Nrwl/Nx, `<rootDir>/node_modules` is not compatible with that.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If the workspace is not monorepo and `jest.config.js` is placed at the root directory, the result is not changed. This should help issues at monorepo.

### CODE REVIEW

**Comment by lacolaco** [comment]
@ahnpnl does the example mean monorepo workspace, right? Is the name `example-app-v12-monorepo` ok?

**Comment by ahnpnl** [comment]
Yes an monorepo project, that name LGTM

File: examples/example-app-v12-monorepo/projects/app2/src/test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// This file is required by karma.conf.js and loads recursively all the .spec and framework files
```
Comment (ahnpnl) [inline examples/example-app-v12-monorepo/projects/app2/src/test.ts:1]: this file belongs to Karma. I think it shouldn't be here.

File: examples/example-app-v12-monorepo/projects/app2/karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// Karma configuration file, see link for more information
```
Comment (ahnpnl) [inline examples/example-app-v12-monorepo/projects/app2/karma.conf.js:1]: this file belongs to Karma. I think it shouldn't be here.

File: examples/example-app-v12-monorepo/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# ExampleAppV12Monorepo
```
Comment (ahnpnl) [inline examples/example-app-v12-monorepo/README.md:1]: Would you please copy the `README` from `example-app-v12`?

File: examples/example-app-v12-monorepo/package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "@angular/router": "~12.0.2",
+    "rxjs": "~6.6.0",
+    "tslib": "^2.1.0",
+    "zone.js": "~0.11.4"
+  },
+  "devDependencies": {
+    "@angular-devkit/build-angular": "~12.0.2",
+    "@angular/cli": "~12.0.2",
+    "@angular/compiler-cli": "~12.0.2",
+    "@types/jasmine": "~3.6.0",
```
Comment (ahnpnl) [inline examples/example-app-v12-monorepo/package.json:30]: This dep should be removed

File: examples/example-app-v12-monorepo/projects/app2/tsconfig.spec.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+/* To learn more about this file see: https://angular.io/config/tsconfig. */
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "../../out-tsc/spec",
+    "types": [
+      "jasmine"
```
Comment (ahnpnl) [inline examples/example-app-v12-monorepo/projects/app2/tsconfig.spec.json:7]: this should be `jest`

File: examples/example-app-v12-monorepo/projects/app2/tsconfig.spec.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    ]
+  },
+  "files": [
+    "src/test.ts",
```
Comment (ahnpnl) [inline examples/example-app-v12-monorepo/projects/app2/tsconfig.spec.json:11]: This should be removed

**Review by ahnpnl — CHANGES_REQUESTED** [review]

**Comment by ahnpnl** [comment]
lint needs fixing with `lint:fix` and snapshot needs to be updated with `yarn test -u`

File: examples/example-app-v12-monorepo/package.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "example-app-v12-monorepo",
+  "version": "0.0.0",
+  "scripts": {
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build",
+    "watch": "ng build --watch --configuration development",
+    "test": "jest -c=projects/app1/jest.config.js",
+    "test-esm": "NODE_OPTIONS=--experimental-vm-modules jest -c=projects/app1/jest-esm.config.js"
```
Comment (ahnpnl) [inline examples/example-app-v12-monorepo/package.json:10]: sadly Windows is not happy with `NODE_OPTIONS`, this option seems to work only with Linux/Mac

**Comment by lacolaco** [comment]
Thank you for your repeatedly kindly reviews.

**Review by ahnpnl — APPROVED** [review]

**Comment by ahnpnl** [comment]
thanks!

**Pull request merged**